src/app/components/categories/categories.component.ts

```diff
@@ -0,0 +1,41 @@
+import {ChangeDetectionStrategy, Component, ViewEncapsulation} from '@angular/core';
+import {BehaviorSubject, combineLatest, map, Observable, of} from 'rxjs';
+import {CategoryModel} from '../../models/category.model';
+import {CategoryService} from '../../services/category.service';
+import {Router} from "@angular/router";
+import {FormControl, Validators} from "@angular/forms";
+
+@Component({
+    selector: 'app-categories',
+    styleUrls: ['./categories.component.scss'],
+    templateUrl: './categories.component.html',
+    encapsulation: ViewEncapsulation.Emulated,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class CategoriesComponent {
+    private readonly _ascFilter = "A-Z";
+    private readonly _descFilter = "Z-A"
+    private _data$: Observable<CategoryModel[]> = this._categoryService.getAll();
+    public sortingData$: Observable<string[]> = of([this._ascFilter, this._descFilter]);
+    public sortingForm = new FormControl(this._ascFilter, [Validators.required])
```

Comment:
Basically if you have form, you don't need subject. You can use sortingForm.valueChanges